src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -175,12 +174,18 @@ public static void init()
           });
 
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(FURNACE_USER_NO_ORE),
-          citizen -> citizen.getWorkBuilding() instanceof BuildingSmeltery && IColonyManager.getInstance().getCompatibilityManager()
-                                                                                .getSmeltableOres()
-                                                                                .stream()
-                                                                                .anyMatch(storage -> !((BuildingSmeltery) citizen.getWorkBuilding()).getCopyOfAllowedItems()
-                                                                                                        .getOrDefault(ORE_LIST, new ArrayList<>())
-                                                                                                        .contains(storage)));
+          citizen -> {
+            if (citizen.getWorkBuilding() instanceof BuildingSmeltery)
+            {
+                final List<ItemStorage> oreList = citizen.getWorkBuilding().getModule(GroupedItemListModule.class).map(m -> m.getList(ORE_LIST)).orElse(ImmutableList.of());
+                return IColonyManager.getInstance().getCompatibilityManager()
+                  .getSmeltableOres()
+                  .stream()
+                  .anyMatch(storage -> !oreList.contains(storage));
```

Comment:
You can do this streaigh in the map of the optional instead of doing the orElse() first.

Reply:
I think this is fine. Much more readable then if I nest this within the optional map